### PR TITLE
injecting Auth onto FcmDetailController

### PR DESCRIPTION
**Pull request by mohammedbahja**
policycompass/policycompass#513

### CODE REVIEW

File: app/modules/fcm/controllers/cytoscape.js [hunk 1]
```diff
@@ -1187,9 +1215,19 @@ angular.module('pcApp.fcm.controllers.cytoscapes', [])
...
-        $scope.user = [
-            { FCMModelId: -1 }, { title: '' }, { description: '' }, { keywords: '' }
-        ];
+        if ($.isEmptyObject(data)) {
+            $scope.user = [
+                { FCMModelId: -1 }, { title: '' }, { description: '' }, { keywords: '' }
+            ];
+        }
+        else {
+            $scope.user = {
```
Comment (FabiApfelkern) [inline app/modules/fcm/controllers/cytoscape.js]: @mohammedbahja I'm curious, what is this for a kind of user?

Comment (cbotsikas) [inline app/modules/fcm/controllers/cytoscape.js]: @mohammedbahja as @FabiApfelkern mentioned, that's either a wrong variable name or wrong data. 
Also within the two cases, the user variable has two different data types. in the first case it's an array of objects where on the second case, it's a single object with all the individual properties of the array mentioned before.
To sum it up, this is definitely not a user variable, based on the data you store, and it's completely wrong to have different types stored on the same variable name. You should also fix the code that is using this.
Either way, i'm closing this PR.

**Comment by cbotsikas** [comment]
See inline comments for 2nd commit

**Comment by mohammedbahja** [comment]
@cbotsikas and @FabiApfelkern 

The user variable was already there by the previous developer. It's used to deal with the modal window. I was surprise as well, but I had to update over this inherited code.   
The code below was already there and it's not used by the way, because there is not modal window in the edit mode.

 $scope.user = [
                { FCMModelId: -1 }, { title: '' }, { description: '' }, { keywords: '' }
            ];


I added this  else part
 
            $scope.user = {
                title: data.title, description: data.description, keywords: data.keywords, domains: data.domains
            };


I will remove the unused code and re-push a gain. The most important thing for me now is to update FcmDetailController file in the STAGE server. I don't use my localhost to test the FCM authentication,and I need this file to be uploaded please. 

Thanks

**Pull request closed without merging**